code/modules/fish/fishtank.dm

```diff
@@ -337,6 +346,7 @@
 	var/caught_fish = input("Select a fish to catch.", "Fishing") as null|anything in fish_types_input		//Select a fish from the tank
 	if(fish_count <= 0)
 		to_chat(user, "There are no fish in [src] to catch!")
+		current_fisher = null
```

Comment:
you can move it to right after the `input()`, outside of an `if`, so that you would only need to set it once, and not in each of the `if` branches

Reply:
Good call.
Moved.